front/components/data_source/salesforce/SalesforceOAuthExtractConfig.tsx

```diff
@@ -13,65 +14,18 @@ export function SalesforceOauthExtraConfig({
   setExtraConfig,
   setIsExtraConfigValid,
 }: ConnectorOauthExtraConfigProps) {
-  const [pkceStatus, setPkceStatus] = useState<{
-    url: string;
-    status: "success" | "loading" | "error" | "idle";
-  }>({
-    url: "",
-    status: "idle",
-  });
-
   useEffect(() => {
     async function generatePKCE() {
-      try {
-        setPkceStatus({
-          url: extraConfig.instance_url,
-          status: "loading",
-        });
-        const response = await fetch(
-          `/api/oauth/pkce?domain=${extraConfig.instance_url}`,
-          {
-            method: "GET",
-            headers: {
-              "Content-Type": "application/json",
-            },
-          }
-        );
-        if (!response.ok) {
-          throw new Error("Failed to generate PKCE challenge");
-        }
-        const { code_verifier, code_challenge } = await response.json();
-        setExtraConfig((extraConfig) => ({
-          ...extraConfig,
-          code_verifier,
-          code_challenge,
-        }));
-        setPkceStatus({
-          url: extraConfig.instance_url,
-          status: "success",
-        });
-      } catch (error) {
-        console.error("Error generating PKCE challenge:", error);
-        setPkceStatus({
-          url: extraConfig.instance_url,
-          status: "error",
-        });
-      }
+      const { code_verifier, code_challenge } = await getPKCEConfig();
```

Comment:
Nice that this works -- I was not managing to get this working 100% of the time when I tried (it randomly failed 1 out of 4 times which is why I had used their endpoint)